src/modifySongData.js

```diff
@@ -0,0 +1,84 @@
+function clamp(min, max, val) {
+  return Math.max(
+    min,
+    Math.min(max, val)
+  );
+}
+
+function calculateMean(vals) {
+  let sum = vals.reduce((sum, cur) => sum + cur, 0);
+  return sum / vals.length;
+}
+
+/**
+ * This method takes an input list of energy values, and outputs a new list. THe
+ * methodology for figuring out the new list is to calculate the mean and standard
+ * deviation of the input values (ignoring zeros). We then define a new range
+ * consisting of some number of standard deviations in either direction from the
+ * mean. Old values are then clamped to our new range, and normalized. Finally
+ * we do some smoothing between frames, based on our smooth factor
+ * @param {number[]} energy - A list of energy values for a single frequency band
+ *   (i.e. bass, mid, high) with values ranging from 0, 255
+ * @param {number} numDeviations - How many standard deviations in either direction
+ *   from the mean we want to include in our new range
+ * @param {number} smoothFactor - What percentage of previous frames to average
+ *   into next frame (should be a number between 0 and 1
+ */
+function getFrequencyEnergy(energy, numDeviations=1.5, smoothFactor=0.7) {
+  const nonZero = energy.filter(e => e > 0);
+  const mean = calculateMean(nonZero);
+
+  const stdDev = Math.sqrt(
+    calculateMean(nonZero.map(x => Math.pow(mean - x, 2)))
+  );
+
+  const min = mean - stdDev * numDeviations;
+  const max = mean + stdDev * numDeviations;
+
+  const ret = energy.map(x => (clamp(min, max, x) - min) / (max - min) * 0xff);
+  return smooth(ret, smoothFactor);
+}
+
+/**
+ * Smooths out an array frame by frame. Each new frame consists of smoothFactor
+ * percent of the previous frame and (1-smoothFactor) percent of the current frame
+ * This method mutates the input rg
+ * @param {number[]} rg
+ * @param {number} smoothFactor
+ */
+function smooth(rg, smoothFactor) {
```

Comment:
I'd be interested in seeing what this looks like with and without smoothing, and maybe also seeing a comment explaining why this part of the process is desired.

Reply:
I'll also add a comment to the code, but this is done so that movement doesnt end up being quite as jerky frame by frame (a big complaint from Marina).